File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.cpp
class Solution {
public:
    int findMin(vector<int> &nums) {
        if (nums[0] < nums[nums.size()-1]) return nums[0];
        int l = 0, r = nums.size()-1;
        while (l<=r)
        {
            int mid = (l+r)/2;
            if (mid!=0 && nums[mid-1] > nums[mid]) return nums[mid];
            else if (nums[mid] < nums[r]) r = mid-1;
            else l = mid + 1;
        }
        return nums[r];
    }
};

File: Data Structures & Algorithms/sliding-window-maximum/submission-1.py
class Solution:
    def maxSlidingWindow(self, nums: List[int], k: int) -> List[int]:
        # print(nums)
        deq = deque([nums[0]])
        # print(deq)
        # print()
        res = []
        for i in range(1, k): 
            # print(f"nums[i]: {nums[i]}")
            while deq and nums[i] > deq[-1]:
                deq.pop()
            deq.append(nums[i])
            # print(deq)
        res.append(deq[0])
        # print(f"res: {res}")
        l = 0
        
        # print()
        for r in range(k, len(nums)):
            # print(f"nums[r]: {nums[r]}")
            if deq[0] == nums[l]: deq.popleft()
            while deq and nums[r] > deq[-1]: 
                deq.pop()
            deq.append(nums[r])
            # print(deq)
            res.append(deq[0])
            # print(f"res: {res}")
            # print()
            l+=1
        return res
